File: unit08-lab/namer.rb
# namer.rb tests class Name and its operations
#
# Begun by: Dr. Jump, for CS 315 at King's College.
# Completed by: Robert Lowman
# Date: 11/3/2016
####################################################

require 'test/unit/assertions'   # needed for assert
include Test::Unit::Assertions

class Name

     def initialize(first, middle, last)
       @first, @middle, @last = first, middle, last
     end

     attr_reader :first, :middle, :last

     def fullName()
       @first + " " + @middle + " " + @last
     end

     def print()
       puts fullName()
     end
      
end


def testName
   name = Name.new("John", "Paul", "Jones")

   assert name.first == "John", "first failed"
   assert name.middle == "Paul", "middle failed"
   assert name.last == "Jones", "last failed"
   assert name.fullName == "John Paul Jones", "fullName failed"
   assert name.print == "John Paul Jones", "print failed"

   print "All tests passed!\n"
end

testName

File: unit08-lab/namer.py
## namer.py tests class Name and its operations
##
## Completed by: Robert Lowman
## Date: 12/1/2016

class Name:

    def __init__(self, first, middle, last):
        self.first = first
        self.middle = middle
        self.last = last

    def fullName(self):
        return self.first + " " + self.middle + " " + self.last

    def printName(self):
        print self.fullName()

aName = Name("John", "Paul", "Jones")
assert aName.first == "John", "Invalid first"
assert aName.middle == "Paul", "Invalid middle"
assert aName.last == "Jones", "Invalid last"
assert aName.fullName() == "John Paul Jones", "Invalid full"
aName.printName()
print "All tests passed!"
